src/odemis/acq/stream/_base.py

```diff
@@ -641,7 +641,7 @@ def _updateDRange(self, data=None):
 
         if data is None:
             if self.raw:
-                if isinstance(self.raw, list):
+                if isinstance(self.raw, list) or isinstance(self.raw, model.DataArray):
```

Comment:
Is that still needed? What can't the caller pass a list with a single DataArray in such case?